tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/commands/StateTest.java

```diff
@@ -146,10 +148,11 @@ protected final void doSetUp() {
 		final IHandlerService handlerService = fWorkbench
 				.getService(IHandlerService.class);
 		handlerActivation = handlerService.activateHandler(COMMAND_ID, handler);
+		command.setEnabled(handlerService.getCurrentState()); // Force handler initialization
```

Comment:
This is a problem. Original Eclipse was able to affect the command state immediately upon handler activaton. However, HandlerServiceHandler does not observe the currently active state of subhandlers activation and operates in pull mode.

Reply:
Added integration tests that ensure that toolbar item does indeed subscribe to command state and polls its handlers.